apps/src/studio/studio.js

```diff
@@ -1255,8 +1254,9 @@ function handleActorCollisionsWithCollidableList (
             collidable.beginRemoveElement();
           } else {
             collidable.removeElement();
-            list.splice(i, 1);
+            list.splice(list.indexOf(collidable), 1);
           }
+          activeCollidables.splice(i, 1);
```

Comment:
My solution is to make a filtered list of `activeCollidables` at the beginning of this algorithm and do all our checks against that.  One consequence is that when we remove an item, we have to remove it from both `list` (which is a reference to `Studio.items`) and from `activeCollidables` so subsequent checks in the loop are correct.  Note that we remove from `activeCollidables` whether we are removing the item immediately or just starting its fade - either way, it's no longer active.